core/src/main/scala/org/ensime/core/Indexer.scala

```diff
@@ -24,10 +24,21 @@ class Indexer(
     LineSourcePositionHelper.fromFqnSymbol(hit)(vfs)
   )
 
-  def oldSearchTypes(query: String, max: Int) =
-    index.searchClasses(query, max).filterNot {
-      name => name.fqn.endsWith("$") || name.fqn.endsWith("$class")
-    }.map(typeResult)
+  def oldSearchTypes(query: String, max: Int): List[TypeSearchResult] = {
+    // Remove $/$class from the end.
+    def strip(fqn: String): String = fqn.replaceAll("\\$(class)*$", "")
+
+    import org.ensime.util.list._
+
+    index.searchClasses(query, max)
```

Comment:
You already closed the PR but I want to discuss the return type of `searchClasses`. I think it is much more understandable to have List[ClassDef] or even Vector[ClassDef]. I would be surprised to get Method/Field instances from `searchClasses`, so code below would be simplified.